pkg/services/sqlstore/migrations/annotation_mig.go

```diff
@@ -109,4 +109,24 @@ func addAnnotationMig(mg *Migrator) {
 	//
 	updateEpochSql := "UPDATE annotation SET epoch = (epoch*1000) where epoch < 9999999999"
 	mg.AddMigration("Convert existing annotations from seconds to milliseconds", NewRawSqlMigration(updateEpochSql))
+
+	//
+	// 6.3: Make Regions a single annotiaon
+	//
+	mg.AddMigration("Add epoch_end column", NewAddColumnMigration(table, &Column{
+		Name: "epoch_end", Type: DB_BigInt, Nullable: false, Default: "0",
+	}))
+	mg.AddMigration("Add index for epoch_end", NewAddIndexMigration(table, &Index{
+		Cols: []string{"epoch_end"}, Type: IndexType,
+	}))
+	mg.AddMigration("Make epoch_end the same as epoch", NewRawSqlMigration("UPDATE annotation SET epoch_end = epoch"))
+	mg.AddMigration("Move existing regions to epoch_end", NewRawSqlMigration(
```

Comment:
MySQL integration test are failing and are usually not that great at supporting advanced update clauses. I think we need to use a code migration here instead, see example here:
https://github.com/grafana/grafana/blob/219d711597fb34e53aba065190311af02c887200/pkg/services/sqlstore/migrations/user_mig.go#L118
https://github.com/grafana/grafana/blob/219d711597fb34e53aba065190311af02c887200/pkg/services/sqlstore/migrations/user_mig.go#L127-L155

Probably need to do it in batches for those Grafana instances having a huge amount of annotations.